src/Fixer/Basic/EmptyBodyFixer.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Basic;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class EmptyBodyFixer extends AbstractFixer
+{
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Empty body of class or function must be abbreviated as `{}` and placed on the same line as the previous symbol, separated by a space.',
+            [new CodeSample('<?php function foo(
+    int $x
+)
+{
+}
+')],
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run after ClassDefinitionFixer, CurlyBracesPositionFixer.
+     */
+    public function getPriority(): int
+    {
+        return -1;
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isAnyTokenKindsFound([T_CLASS, T_FUNCTION]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        for ($index = $tokens->count() - 1; $index > 0; --$index) {
+            if (!$tokens[$index]->isGivenKind([T_CLASS, T_FUNCTION])) {
+                continue;
+            }
+
+            $openBraceIndex = $tokens->getNextTokenOfKind($index, ['{', ';']);
+            \assert(\is_int($openBraceIndex));
```

Comment:
Is it project's convention to use `assert()` in the fixers? I didn't see it before.

Reply:
Is it project's convention to NOT use `assert()` in the fixers? If so, I'd like to challenge that as IMHO it is much better than PHPDoc (enforces type being what it claims to be, not only promises that).

Reply:
It looks like some ADR to be made, I wouldn't introduce such new approach in PR with new fixer, but rather use [RFC](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/discussions/categories/rfc) discussion for that 🙂. To be honest I have never used native `assert()` and I don't even know how it would behave here (from user perspective in CLI). I am 100% for such improvements, but I think it's not good moment to introduce it 😉.